pilot/pkg/proxy/envoy/v2/pushqueue.go

```diff
@@ -56,27 +93,20 @@ func (p *PushQueue) Enqueue(proxy *XdsConnection, pushInfo *PushEvent) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
-	event, exists := p.eventsMap[proxy]
-	if !exists {
-		p.eventsMap[proxy] = pushInfo
-		p.connections = append(p.connections, proxy)
-	} else {
-		event.push = pushInfo.push
-		event.full = event.full || pushInfo.full
-		// When full push, do not care about edsUpdatedServices
-		if !event.full {
-			edsUpdates := map[string]struct{}{}
-			for endpoint := range pushInfo.edsUpdatedServices {
-				edsUpdates[endpoint] = struct{}{}
-			}
-			for endpoint := range event.edsUpdatedServices {
-				edsUpdates[endpoint] = struct{}{}
-			}
-			event.edsUpdatedServices = edsUpdates
-		} else {
-			event.edsUpdatedServices = nil
-		}
+	// If its already in progress, merge the info and return
```

Comment:
but you are not going to reconstruct the push context. So you end up losing an update. ?

Reply:
I don't think we lose an update. Rather, we skip one. Instead of going A -> B -> C we go A -> C. This should be ok. What is a problem is if we go A -> C -> B (possible without this change) or A and never send B or C.